### PR TITLE
25 Tag Entry Prompt

### PR DESCRIPTION
**Pull request by JikoJeans**
# Description

This pull request seeks to add in functional requirement UX13 tag entry prompt. To fulfill this a stateful component is made that will be used to hold all the other components together. There must be a search bar that filters the tags that are present and if no compatible tags are found then the widget should display a create tag button. Additionally if the field is submitted then the tag will be added if the tag is not already in the list.

Fixes # 25

## Type of change

Please delete options that are not relevant.

- [ ] New feature (non-breaking change which adds functionality)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A - unit test
       - The unit test is going to cover testing the filter functionality, creation/deletion of tags, and if the page can be accessed from within the app. These can be tested using the command 'flutter test integration_test --branch-coverage'  and locally on my end all test are passing.
- [ ] Test B - when using the widget you should be able to type in any desired tag and it should filter in real time along with update new tags in real time. I will attempt to record a video showing this.

https://github.com/S1robe/Pocket-Therapist/assets/143761091/b82a6431-3050-48d1-b4c6-50a5f5b0a2ab

**Test Configuration**:
* Firmware version: Andriod v14
* Hardware: Pixel 7
* Toolchain: andriod API 34
* SDK: sdk_gphone64_x86_64

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/lib/pages/settings_tag.dart [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+    'Centered',
+    'Content',
+    'Fulfilled',
+    'Patient',
+    'Peaceful',
+    'Present',
+    'Relaxed',
+    'Serene',
+    'Trusting',
+  ];
```
Comment (S1robe) [inline app/lib/pages/settings_tag.dart:27]: You can actually add this now.
You can specify these as the default, if nothing is found when loading. Then add a section in settings under the save() and load() methods to either call your settings, or the modify this as a global varibles.

Ideally this tag list would be stored in settings so we dont have to do any visibility trickery with statics or imports.

lib/providers/settings.dart

**Review by S1robe — CHANGES_REQUESTED** [review]
Please establish persistence for the tags. We can do that by mimicking the behavior in settings.

File: app/lib/pages/settings_tag.dart (hunk 1 above)
Comment (JikoJeans) [reply app/lib/pages/settings_tag.dart]: Copy that, the tag list has been moved to settings and the methods save, load, and default have been updated. Thank you for the advice it made the app more scalable!

**Review by S1robe — APPROVED** [review]
Awesome job, persistence was added. 
Thank you for your contribution.

**Review by Benek6h — APPROVED** [review]
Looks great! Thank you for getting this done so quickly!

**Pull request merged**